source/vir/src/context.rs

```diff
@@ -198,6 +198,25 @@ impl GlobalCtx {
         for t in &krate.traits {
             crate::recursive_types::add_trait_to_graph(&mut func_call_graph, t);
         }
+        for f in &krate.functions {
+            // Heuristic: add all external_body functions first.
+            // This is currently needed because:
+            // - external_body broadcast_forall functions are currently implicitly imported
+            // - open_atomic_invariant_begin/open_local_invariant_begin/open_invariant_end
```

Comment:
`open_atomic_invariant_begin` and friends are all marked external, so they should be completely ignored, right? Why do we need to account for them?

Reply:
Oops, my comment was inaccurate.  I misdiagnosed what was going on.